frontend/src/stimulus/controllers/dynamic/meetings/check-unsaved.controller.ts

```diff
@@ -44,6 +46,7 @@ export default class extends ApplicationController {
 
     const context = await window.OpenProject.getPluginContext();
     this.turboRequests = context.services.turboRequests;
+    this.beforeUnloadController = this.application.getControllerForElementAndIdentifier(document.body, 'beforeunload') as BeforeunloadController;
```

Comment:
Can we use [Stimulus outlets](https://stimulus.hotwired.dev/reference/outlets) here instead? (@akabiru is the expert)

Reply:
For me, an outlet really only is useful in a dynamic setting. In this case, we reference a global controller on the body. I find neither the outlet nor the TS mixins more readable than a simple global. Maybe we leave it as a global then if you object to the `getControllerForElementAndIdentifier` pattern?

Reply:
I think leaving it as a (plain old) global object is also fine. However if we do go with a Stimulus controller then I'd personally prefer we go the whole hog, and use one of the above aforementioned patterns for composability.

Reply:
As we discussed in our call, this is what I did not and it feels a lot better to keep it that way (+ the PageState you suggested). I'll let you have a final pass over this PR